samples/dnn/object_tracker.cpp

```diff
@@ -0,0 +1,355 @@
+            return 2;
+        }
+    }
+    else if (inputName.size())
+    {
+        string filePath = findFile(inputName);
+        if (!cap.open(filePath))
+        {
+            cout << "Could not open: " << inputName << endl;
+            return 2;
+        }
+    }
+
+    Mat image;
+
+    for (;;)
+    {
+        cap >> image;
+        if (image.empty())
+        {
+            cerr << "Can't capture frame. End of video stream?" << endl;
+            return 2;
+        }
+        else if (imgWidth == -1){
+            imgWidth = min(image.rows, image.cols);
+            fontSize = (stdSize*imgWidth)/stdImgSize;
+            fontWeight = (stdWeight*imgWidth)/stdImgSize;
+        }
+        const string label = "Press space bar to pause video to draw bounding box.";
+        Rect r = getTextSize(Size(), label, Point(), fontFace, fontSize, fontWeight);
+        r.height += 2 * fontSize; // padding
+        r.width += 10; // padding
+        rectangle(image, r, Scalar::all(255), FILLED);
+        putText(image, label, Point(10, fontSize), Scalar(0,0,0), fontFace, fontSize, fontWeight);
+        putText(image, "Press space bar after selecting.", Point(10, 2*fontSize), Scalar(0,0,0), fontFace, fontSize, fontWeight);
+        imshow(windowName, image);
+        int key = waitKey(200);
+        if (key == ' ')
+        {
+            selectRect = selectROI(windowName, image);
+            break;
+        }
+        if (key == 27) // ESC key to exit
+        {
+            exit(0);
+        }
+    }
+
+    Mat image_select = image.clone();
+
+    cout << "ROI=" << selectRect << endl;
+
+    tracker->init(image, selectRect);
+
+    TickMeter tickMeter;
+
+    for (int count = 0; ; ++count)
+    {
+        cap >> image;
+        if (image.empty())
+        {
+            cerr << "Can't capture frame " << count << ". End of video stream?" << endl;
+            break;
+        }
+
+        Rect rect;
+
+        tickMeter.start();
+        bool ok = tracker->update(image, rect);
+        tickMeter.stop();
+
+        float score = getTrackingScore(tracker);
+
+        Mat render_image = image.clone();
+
+        if (ok)
+        {
+            rectangle(render_image, rect, Scalar(0, 255, 0), 2);
+
+            string timeLabel = format("Inference time: %.2f ms", tickMeter.getTimeMilli());
+            string scoreLabel = format("Score: %f", score);
+            Rect r = getTextSize(Size(), timeLabel, Point(), fontFace, fontSize, fontWeight);
+            r.height += 2 * fontSize; // padding
+            r.width += 10; // padding
+            rectangle(render_image, r, Scalar::all(255), FILLED);
+            putText(render_image, timeLabel, Point(10, fontSize), Scalar(0,0,0), fontFace, fontSize, fontWeight);
+            putText(render_image, scoreLabel, Point(10, 2*fontSize), Scalar(0,0,0), fontFace, fontSize, fontWeight);
+       }
+
+        imshow(windowName, render_image);
+
+        tickMeter.reset();
+
+        int c = waitKey(100);
```

Comment:
Why do you use 100 here?

Reply:
yes, 100ms is too long delay. Should be decreased to 10.